crates/storage/db/src/abstraction/table.rs

```diff
@@ -35,6 +35,11 @@ pub trait Compress: Send + Sync + Sized + Debug {
 pub trait Decompress: Send + Sync + Sized + Debug {
     /// Decompresses data coming from the database.
     fn decompress<B: AsRef<[u8]>>(value: B) -> Result<Self, DatabaseError>;
+
+    /// Decompresses owned data coming from the database.
+    fn decompress_owned(value: Vec<u8>) -> Result<Self, DatabaseError> {
+        Self::decompress(value)
+    }
```

Comment:
hmm, do we need it? Intuitively `decompress` seems enough for both cases because `Vec<u8>` implements `AsRef<[u8]>`

Reply:
On `main`, `RawValue` is creating a copy of value with `value.as_ref().to_vec()`. 

But if `decode` gives out a `Cow::Owned(Vec<u8>)`, I'd rather `RawValue` get the actual value instead of duplicating it. I don't know if rust optimizes it, that's why I'm explicitly defining it as such